service/src/main/java/bio/terra/cbas/common/MicrometerMetrics.java

```diff
@@ -1,41 +1,121 @@
 package bio.terra.cbas.common;
 
+import static bio.terra.cbas.common.JavaMethodUtil.loggableMethodName;
+
 import bio.terra.cbas.models.CbasRunStatus;
 import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.ImmutableTag;
 import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
 import io.micrometer.core.instrument.Timer;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
 import org.springframework.stereotype.Component;
 
 @Component
 public class MicrometerMetrics {
 
   private final MeterRegistry meterRegistry;
 
+  private static final String STATUS_TAG = "status";
+  private static final String RESPONSE_CODE_TAG = "response_code";
+
   public MicrometerMetrics(MeterRegistry meterRegistry) {
     this.meterRegistry = meterRegistry;
   }
 
-  public void logRunCallback(CbasRunStatus resultsStatus) {
-    Counter counter =
-        Counter.builder("run_callback")
-            .tag("status", resultsStatus.toString())
-            .register(meterRegistry);
-    counter.increment();
+  /* ******** Counter metrics ******** */
+
+  public void recordRunCallback(CbasRunStatus resultsStatus) {
+    recordCounterMetric(
+        "run_callback", List.of(new ImmutableTag(STATUS_TAG, resultsStatus.toString())), 1);
+  }
+
+  public void recordRunStatusUpdate(CbasRunStatus resultsStatus) {
+    recordCounterMetric(
+        "run_smartpoller_update",
```

Comment:
your addition of the tag constants made me think: should the metric names be constants too? then they'd be all in one place for us to look up when we want to make queries in the azure portal